### PR TITLE
Drop Python 3.7 support

### PR DESCRIPTION
**Pull request by rominf**
# About this change: What it does, why it matters

Python 3.7 recently reached its end-of-life:
https://devguide.python.org/versions/#versions

### CODE REVIEW

**Review by bugant — APPROVED** [review]

**Review by bugant — APPROVED** [review]

File: .github/workflows/build.yaml [hunk 1]
```diff
@@ -52,7 +52,7 @@ jobs:
     needs: lint
     strategy:
       matrix:
-        python-version: ['3.7', '3.8', '3.9', '3.10', '3.11', 'pypy-3.7', 'pypy-3.8', 'pypy-3.9']
+        python-version: ['3.8', '3.9', '3.10', '3.11', 'pypy-3.8', 'pypy-3.9']
```
Comment (Mulugruntz) [inline .github/workflows/build.yaml:55]:
```suggestion
        python-version: ['3.8', '3.9', '3.10', '3.11', 'pypy-3.9', 'pypy-3.10']
```

It seems that officially, pypy is on 3.9 and 3.10: https://www.pypy.org/download_advanced.html

We could also decide for pypy to leave 3.8 and add 3.10 (3.8, 3.9, 3.10).

File: pyproject.toml [hunk 2]
```diff
@@ -66,7 +65,7 @@ version-file = "aiven/client/version.py"
 
 [tool.black]
 line-length = 125
-target-version = ['py37', 'py38', 'py39']  # 'py310' is not yet available on Fedora 34
+target-version = ['py38', 'py39']  # 'py310' is not yet available on Fedora 34
```
Comment (Mulugruntz) [inline pyproject.toml:68]: Not Fedora34 anymore.
Can add
```suggestion
target-version = ['py38', 'py39', 'py310', 'py311']
```

File: .github/workflows/build.yaml (hunk 1 above)
Comment (rominf) [reply .github/workflows/build.yaml:55]: Thanks, I noticed that. I wanted to put this into separate PR to make PR and commit history clean.

File: pyproject.toml (hunk 2 above)
Comment (rominf) [reply pyproject.toml:68]: Thanks, I noticed that. I wanted to put this into separate PR to make PR and commit history clean.

Comment (Mulugruntz) [reply pyproject.toml:68]: As much as the pypy line could make sense to have in a separate PR, I think this one here is part of this PR.
You can amend + force push to have a clean history.

Comment (rominf) [reply pyproject.toml:68]: I still think that it should go into separate PR, as it requires bumping requirements in `pyproject.toml` file (`black==23.3.0`). It's better than both: PR and commits are atomic, it makes easier to reason about the changes in the future.

Comment (Mulugruntz) [reply pyproject.toml:68]: Okay, let's do that in another PR then!

**Review by Mulugruntz — APPROVED** [review]

**Comment by Mulugruntz** [comment]
@rominf I can merge, but ideally, it'd be great to have an internal ticket number and put it in the PR description beforehand.
WDYT?

**Review by ngilles-aiven — APPROVED** [review]

**Comment by ngilles-aiven** [comment]
@Mulugruntz oops, didn't see your last comment there. I don't think it's super important to have an internal Jira number in this case... in fact aiven-client is public project (we allow external contributors), and we used to not put internal numbers in there at all. At some point we gave up on that and added them anyway for Jira integration to work.

**Pull request merged**